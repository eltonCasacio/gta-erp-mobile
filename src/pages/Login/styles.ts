import styled from 'styled-components/native';

export const Wrapper = styled.View`
  justify-content: space-between;
  align-items: center;
  flex: 1;
`;

export const InputContainer = styled.View`
  width: 100%;
`;
