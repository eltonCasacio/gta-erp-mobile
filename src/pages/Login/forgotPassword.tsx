import React, {useState} from 'react';
import * as S from './styles';

import Logo from '../../components/Logo';
import Input from '../../components/Input';
import Buttom from '../../components/Button';

const ForgotPassword = ({navigation}: any) => {
  const [email, setEmail] = useState('');

  const requestPassword = async () => {
    console.log('EMAIL PARA RESET DE SENHA:', email);
    navigation.goBack();
  };

  return (
    <S.Wrapper>
      <Logo />
      <S.InputContainer>
        <Input
          placeholder="email"
          onChangeText={(text: any) => setEmail(text)}
        />
        <S.Label>
          Caso n√£o possua e-mail nem whatsapp, favor procurar o RH
        </S.Label>
      </S.InputContainer>
      <Buttom label="SOLICITAR" callback={requestPassword} />
    </S.Wrapper>
  );
};

export default ForgotPassword;
