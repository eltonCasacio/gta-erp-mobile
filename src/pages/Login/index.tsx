import React from 'react';
import {Wrapper, InputContainer} from './styles';

import Logo from '../../components/Logo';
import Input from '../../components/Input';
import Buttom from '../../components/Button';

const Login = () => {
  const exec = () => {
    console.log('teste');
  };
  return (
    <Wrapper>
      <Logo />
      <InputContainer>
        <Input placeholder="usuÃ¡rio" />
        <Input placeholder="senha" />
      </InputContainer>
      <Buttom label="ENTRAR" callback={exec} />
    </Wrapper>
  );
};

export default Login;
