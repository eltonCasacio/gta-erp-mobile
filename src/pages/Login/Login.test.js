import React from 'react';
import renderer from 'react-test-renderer';

import Login from '.';

describe('<Login />', () => {
  it('should render default layout', () => {
    const tree = renderer.create(<Login />).toJSON();

    //tree.children[2].props.onClick();

    expect(tree).toMatchSnapshot();
  });
});
