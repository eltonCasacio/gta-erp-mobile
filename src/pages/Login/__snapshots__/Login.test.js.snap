// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render default layout 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <Image
    accessibilityLabel="Logo da GTA"
    source={
      Object {
        "testUri": "../../../src/assets/logo.png",
      }
    }
    style={
      Array [
        Object {},
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fafafa",
            "borderRadius": 3,
            "marginBottom": 5,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        placeholder="usuÃ¡rio"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#070E70",
              "fontSize": 22,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fafafa",
            "borderRadius": 3,
            "marginBottom": 5,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        placeholder="senha"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#070E70",
              "fontSize": 22,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5762F2",
        "borderRadius": 200,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "width": 200,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fafafa",
            "fontSize": 22,
          },
        ]
      }
    >
      ENTRAR
    </Text>
  </View>
</View>
`;
